test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpLogExporterTests.cs

```diff
@@ -25,51 +28,162 @@ namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests
 {
     public class OtlpLogExporterTests : Http2UnencryptedSupportTests
     {
-        [Fact]
-        public void ToOtlpLogRecordTest()
-        {
-            // Just a basic test to demonstrate an
-            // approach useful for testing.
-            // This needs to be expanded to
-            // actually test the conversion.
-            List<LogRecord> logRecords = new List<LogRecord>();
+        private readonly ILogger logger;
+        private readonly List<LogRecord> exportedItems = new();
+        private readonly ILoggerFactory loggerFactory;
+        private OpenTelemetryLoggerOptions options;
 
-            using var loggerFactory = LoggerFactory.Create(builder =>
+        public OtlpLogExporterTests()
+        {
+            this.loggerFactory = LoggerFactory.Create(builder =>
             {
                 builder.AddOpenTelemetry(options =>
                 {
-                    options.IncludeFormattedMessage = true;
-                    options.ParseStateValues = true;
-                    options.AddInMemoryExporter(logRecords);
+                    this.options = options;
+                    this.options.AddInMemoryExporter(this.exportedItems);
                 });
             });
 
-            // TODO:
-            // Validate attributes, severity, traceid,spanid etc.
+            this.logger = this.loggerFactory.CreateLogger<OtlpLogExporterTests>();
+        }
+
+        [Fact]
+        public void CheckToOtlpLogRecordStateValues()
+        {
+            this.options.IncludeFormattedMessage = true;
```

Comment:
There are cases in `LogRecordTest.cs` demonstrating this: https://github.com/open-telemetry/opentelemetry-dotnet/blob/4323c93e14c1aa56e9f81717321818786d80ccd7/test/OpenTelemetry.Tests/Logs/LogRecordTest.cs#L488
So modifying options after the factory was built is not a recommended usage?

Reply:
I see . Thanks for sharing the pointers. 
Yes logging config is bit different, and would be good doc additions.

Reply:
I think by default we want to avoid updates on options after ctor, because this would introduce complexity which makes the code difficult to maintain - and in many cases, we will see things ended up in "undefined behavior".

In certain cases we may explicitly allow dynamic configuration update after ctor, these need to be carefully designed, implemented and documented. For example - https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/src/OpenTelemetry#processor.

For this specific case, I don't recall if there is a scenario that needs the dynamic configuration update.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/issues/2902